### PR TITLE
feat(tables): update samples to show explainability

### PR DESCRIPTION
**Pull request by sirtorry**
- adds global feature importance to get_model()
- adds local feature importance to predict()

### CODE REVIEW

File: tables/automl/automl_tables_model.py [hunk 1]
```diff
@@ -156,12 +167,24 @@ def get_model(project_id, compute_region, model_display_name):
     else:
         deployment_state = "undeployed"
 
+    # get features of top importance
+    feat_list = [
+        (x.feature_importance, x.column_display_name)
```
Comment (leahecole) [inline tables/automl/automl_tables_model.py]: Can we use `feature` or something similar instead of `x` pretty please?

Comment (sirtorry) [reply tables/automl/automl_tables_model.py]: will do

File: tables/automl/predict_test.py [hunk 2]
```diff
@@ -43,15 +43,15 @@ def test_predict(capsys):
...
     }
 
     ensure_model_online()
-    automl_tables_predict.predict(PROJECT, REGION, STATIC_MODEL, inputs)
+    automl_tables_predict.predict(PROJECT, REGION, STATIC_MODEL, inputs, True)
     out, _ = capsys.readouterr()
-    assert 'Predicted class name:' in out
-    assert 'Predicted class score:' in out
+    assert "Predicted class name:" in out
+    assert "Predicted class score:" in out
```
Comment (leahecole) [inline tables/automl/predict_test.py]: should we add an assertion for the feature importance too?

Comment (sirtorry) [reply tables/automl/predict_test.py]: sure. although these tests don't do much.

Comment (leahecole) [reply tables/automl/predict_test.py]: Ty - it's more to be setting a good example and displaying good practices imho

**Comment by sirtorry** [comment]
update to use this: https://github.com/googleapis/google-cloud-python/pull/9816

File: tables/automl/predict_test.py (hunk 2 above)
Comment (sirtorry) [reply tables/automl/predict_test.py]: important note: feature importance might not be in _every_ predict call

**Comment by sirtorry** [comment]
@leahecole PTAL and merge when ready. kokoro seems to be failing for reasons not related to this PR.

File: tables/automl/automl_tables_dataset.py [hunk 3]
```diff
@@ -79,23 +79,38 @@ def list_datasets(project_id, compute_region, filter_=None):
...
-        print("Dataset target column spec id: {}".format(
-            metadata.target_column_spec_id))
-        print("Dataset target column spec id: {}".format(
-            metadata.target_column_spec_id))
-        print("Dataset weight column spec id: {}".format(
-            metadata.weight_column_spec_id))
-        print("Dataset ml use column spec id: {}".format(
-            metadata.ml_use_column_spec_id))
+        print(
+            "Dataset primary table spec id: {}".format(
```
Comment (gguuss) [inline tables/automl/automl_tables_dataset.py]: (Optionally fix this) I'll just note that I prefer single-quotes over double-quotes and that our authoring guide suggests single quotes.

File: tables/automl/automl_tables_dataset.py [hunk 4]
```diff
@@ -79,23 +79,38 @@ def list_datasets(project_id, compute_region, filter_=None):
...
+                metadata.primary_table_spec_id
+            )
+        )
+        print(
```
Comment (gguuss) [inline tables/automl/automl_tables_dataset.py]: This looks like the output from black, feel free to ignore my comments if this is now out of our control but this style gives me a knee-jerk reaction. I don't prefer this paren-wrapped code block style. The following shows how I would have done this:

```
print('Dataset target column spec id: {}.format(
    metadata.weight_column_spec_id))
```

File: tables/automl/automl_tables_dataset.py [hunk 5]
```diff
@@ -288,7 +308,7 @@ def import_data(project_id, compute_region, dataset_display_name, path):
     client = automl.TablesClient(project=project_id, region=compute_region)
 
     response = None
-    if path.startswith('bq'):
+    if path.startswith("bq"):
```
Comment (gguuss) [inline tables/automl/automl_tables_dataset.py]: Again, I prefer single-quotes over double quotes and this is what's in our Python style guide.

File: tables/automl/automl_tables_dataset.py [hunk 6]
```diff
@@ -297,7 +317,7 @@ def import_data(project_id, compute_region, dataset_display_name, path):
         input_uris = path.split(",")
         response = client.import_data(
             dataset_display_name=dataset_display_name,
-            gcs_input_uris=input_uris
+            gcs_input_uris=input_uris,
```
Comment (gguuss) [inline tables/automl/automl_tables_dataset.py]: This looks like more blackening ugliness. I'm not sure the comma is necessary.

File: tables/automl/automl_tables_dataset.py [hunk 7]
```diff
@@ -331,12 +353,14 @@ def export_data(project_id, compute_region, dataset_display_name, gcs_uri):
...
-                   target_column_spec_name=None,
-                   weight_column_spec_name=None,
-                   test_train_column_spec_name=None):
+def update_dataset(
+    project_id,
+    compute_region,
+    dataset_display_name,
+    target_column_spec_name=None,
+    weight_column_spec_name=None,
+    test_train_column_spec_name=None,
```
Comment (gguuss) [inline tables/automl/automl_tables_dataset.py]: Why the trailing comma here?

File: tables/automl/automl_tables_model.py [hunk 8]
```diff
@@ -116,19 +118,28 @@ def list_models(project_id, compute_region, filter_=None):
         print("Model id: {}".format(model.name.split("/")[-1]))
         print("Model display name: {}".format(model.display_name))
         metadata = model.tables_model_metadata
-        print("Target column display name: {}".format(
-            metadata.target_column_spec.display_name))
-        print("Training budget in node milli hours: {}".format(
-            metadata.train_budget_milli_node_hours))
-        print("Training cost in node milli hours: {}".format(
-            metadata.train_cost_milli_node_hours))
+        print(
```
Comment (gguuss) [inline tables/automl/automl_tables_model.py]: This looks weird. Why the `print(` on its own line instead of minimizing the vertical space used?

File: tables/automl/automl_tables_model.py [hunk 9]
```diff
@@ -156,12 +167,24 @@ def get_model(project_id, compute_region, model_display_name):
     else:
         deployment_state = "undeployed"
 
+    # get features of top importance
+    feat_list = [
```
Comment (gguuss) [inline tables/automl/automl_tables_model.py]: This looks like it's actually generating an array of tuples, maybe update the comment above. Does the for loop have to come after the tuple syntax?

File: tables/automl/automl_tables_model.py [hunk 10]
```diff
@@ -304,12 +329,15 @@ def display_evaluation(
     regression_metrics = model_evaluation.regression_evaluation_metrics
     if str(regression_metrics):
         print("Model regression metrics:")
-        print("Model RMSE: {}".format(
-          regression_metrics.root_mean_squared_error
-        ))
+        print(
```
Comment (gguuss) [inline tables/automl/automl_tables_model.py]: This print syntax looks weird.

File: tables/automl/automl_tables_model.py [hunk 11]
```diff
@@ -304,12 +329,15 @@ def display_evaluation(
...
+            "Model RMSE: {}".format(regression_metrics.root_mean_squared_error)
+        )
         print("Model MAE: {}".format(regression_metrics.mean_absolute_error))
-        print("Model MAPE: {}".format(
-            regression_metrics.mean_absolute_percentage_error))
+        print(
```
Comment (gguuss) [inline tables/automl/automl_tables_model.py]: I prefer the previous style, much easier to read.

File: tables/automl/automl_tables_model.py [hunk 12]
```diff
@@ -391,7 +419,7 @@ def delete_model(project_id, compute_region, model_display_name):
     create_model_parser.add_argument("--dataset_display_name")
     create_model_parser.add_argument("--model_display_name")
     create_model_parser.add_argument(
-        "--train_budget_milli_node_hours", type=int,
+        "--train_budget_milli_node_hours", type=int
```
Comment (gguuss) [inline tables/automl/automl_tables_model.py]: Why did you add trailing commas elsewhere and remove them here?

File: tables/automl/automl_tables_predict.py [hunk 13]
```diff
@@ -41,23 +44,50 @@ def predict(project_id,
...
-        print("Predicted class name: {}".format(result.display_name))
-        print("Predicted class score: {}".format(
-            result.classification.score))
+        print(
+            "Predicted class name: {}".format(result.tables.value.string_value)
+        )
+        print("Predicted class score: {}".format(result.tables.score))
+
+        if feature_importance:
+            # get features of top importance
```
Comment (gguuss) [inline tables/automl/automl_tables_predict.py]: If you changed the other comment / code block, change this one as well.

File: tables/automl/dataset_test.py [hunk 14]
```diff
@@ -91,8 +87,8 @@ def test_dataset_update(capsys):
         PROJECT,
         REGION,
         dataset.display_name,
-        target_column_spec_name='Deposit',
-        weight_column_spec_name='Balance'
+        target_column_spec_name="Deposit",
+        weight_column_spec_name="Balance",
```
Comment (gguuss) [inline tables/automl/dataset_test.py]: The trailing comma on the parameter is weird.

File: tables/automl/dataset_test.py [hunk 15]
```diff
@@ -91,8 +87,8 @@ def test_dataset_update(capsys):
         PROJECT,
         REGION,
         dataset.display_name,
-        target_column_spec_name='Deposit',
-        weight_column_spec_name='Balance'
+        target_column_spec_name="Deposit",
```
Comment (gguuss) [inline tables/automl/dataset_test.py]: Any reason for switching from single-quotes to double quotes?
